resources/interactive.mjs

```diff
@@ -0,0 +1,186 @@
+        checkbox.onchange();
+        checkboxes.push(checkbox);
+
+        li.appendChild(checkbox);
+        var label = document.createElement('label');
+        label.appendChild(document.createTextNode(formatTestName(suite.name)));
+        li.appendChild(label);
+        label.htmlFor = checkbox.id;
+
+        const testList = document.createElement('ol');
+        for (let testIndex = 0; testIndex < suite.tests.length; testIndex++) {
+            const testItem = document.createElement('li');
+            const test = suite.tests[testIndex];
+            const anchor = document.createElement('a');
+            anchor.id = suite.name + '-' + test.name;
+            test.anchor = anchor;
+            anchor.appendChild(document.createTextNode(formatTestName(suite.name, test.name)));
+            testItem.appendChild(anchor);
+            testList.appendChild(testItem);
+        }
+        li.appendChild(testList);
+
+        ol.appendChild(li);
+    }
+
+    control.appendChild(ol);
+
+    let button = document.createElement('button');
+    button.textContent = 'Step';
+    button.onclick = onStep;
+    control.appendChild(button);
+
+    button = document.createElement('button');
+    button.textContent = 'Run';
+    button.id = 'runSuites';
+    button.onclick = onRunSuites;
+    control.appendChild(button);
+
+    button = document.createElement('button');
+    button.textContent = 'Select all';
+    button.onclick = () => {
+        for (var suiteIndex = 0; suiteIndex < suites.length; suiteIndex++) {
+            suites[suiteIndex].disabled = false;
+            checkboxes[suiteIndex].checked = true;
+        }
+    };
+    control.appendChild(button);
+
+    button = document.createElement('button');
+    button.textContent = 'Unselect all';
+    button.onclick = () => {
+        for (var suiteIndex = 0; suiteIndex < suites.length; suiteIndex++) {
+            suites[suiteIndex].disabled = true;
+            checkboxes[suiteIndex].checked = false;
+        }
+
+    };
+    control.appendChild(button);
+
+    return control;
+}
+
+const parseQueryString = (function (pairList)
+{
+    const pairs = {};
+    for (let i = 0; i < pairList.length; ++i) {
+        const keyValue = pairList[i].split('=', 2);
+        if (keyValue.length == 1)
+            pairs[keyValue[0]] = '';
+        else
+            pairs[keyValue[0]] = decodeURIComponent(keyValue[1].replace(/\+/g, ' '));
+    }
+    return pairs;
+})(window.location.search.substr(1).split('&'));
+
+function disableAllSuitesExcept(suiteName)
+{
+    Suites.forEach(function(element) {
+        if (element.name !== suiteName)
+            element.disabled = true;
+    });
+}
+
+function startTest()
+{
+    const queryParam = parseQueryString['suite'];
+    if (queryParam !== undefined)
+        disableAllSuitesExcept(queryParam);
+
+    const benchmarkClient = {
+        stepperPromise: undefined,
+        stepper: undefined,
+        step() {
```

Comment:
Nit: maybe rename this to `enableStepping`? It's a bit awkward reading this right now.

Reply:
refactored the code to be easier to undestand.